README.md

```diff
@@ -324,6 +325,63 @@ You can automatically reformat your commit via:
 $ cargo fmt --all
 ```
 
+## Non-Consensus Breaking Release Process
+
+For non-consensus breaking releases, this project uses the following release process:
+
+1. A release manager enumerates the PRs or issues that would _block_
+   the release. A label should be applied to each such issue/PR as
+   `2.1.x.y.z-blocker`. The release manager should ping these
+   issue/PR owners for updates on whether or not those issues/PRs have
+   any blockers or are waiting on feedback.
+
+1. Once all `2.1.x.y.z-blocker` PRs have been merged to `develop`, the release
+   manager will open a `develop -> master` PR.
+
+1. Immediately following this, the release manager will then create a new tag
+   by manually triggering the [`stacks-blockchain` Github Actions workflow](https://github.com/blockstack/stacks-blockchain/actions/workflows/stacks-blockchain.yml)
+   against the `develop` branch, inputting the release candidate tag, `2.1.x.y.z-rc0`,
+   in the Action's input textbox.
+
+1. Once the release candidate has been built, and docker images, etc. are available,
+   the release manager will notify various ecosystem participants to test the release
+   candidate on various staging infrastructure:
+
+   1. Stacks Foundation staging environments.
+   1. Hiro PBC regtest network.
+   1. Hiro PBC testnet network.
+   1. Hiro PBC mainnet mock miner.
+
+   The release candidate should be announced in the `#stacks-core-devs` channel in the
+   Stacks Discord. For coordinating rollouts on specific infrastructure, the release
+   manager should contact the above participants directly either through e-mail or
+   Discord DM. The release manager should also confirm that the built release on the
+   [Github releases](https://github.com/blockstack/stacks-blockchain/releases/)
+   page is marked as `Pre-Release`.
+
+1. The release manager will test that the release candidate successfully syncs with
+   the current chain from genesis both in testnet and mainnet. This requires starting
+   the release candidate with an empty chainstate and confirming that it synchronizes
+   with the current chain tip.
+
+1. If bugs or issues emerge from the rollout on staging infrastructure, the release
+   will be delayed until those regressions are resolved. As regressions are resolved,
+   additional release candidates should be tagged. The release manager is responsible
+   for updating the `develop -> master` PR with information about the discovered issues,
+   even if other community members and developers may be addressing the discovered
+   issues.
+
+1. Once the final release candidate has rolled out successfully without issue on the
+   above staging infrastructure, the release manager tags 2 additional `stacks-blockchain`
+   team members to review the `develop -> master` PR.
```

Comment:
From experience, I find useful to draft that PR early on (one of the very first steps). It helps the release manager getting a good understanding of the deltas for the incoming release, and target testing on specific areas. WDYT?

Reply:
Yeah, I think that's a good idea, will add.